posthog/queries/util.py

```diff
@@ -19,9 +19,18 @@ class PersonPropertiesMode(Enum):
     USING_PERSON_PROPERTIES_COLUMN = auto()
     # Used for generating query on Person table
     DIRECT = auto()
+    """Get person property from the persons table, selecting the latest version of the person."""
+    DIRECT_ON_PERSONS = auto()
+    """
+    Get person property from the persons table WITHOUT aggregation by version. Not fully accurate, as old versions
+    of the person will be matched, but useful for prefiltering on whether _any_ version of the person has ever matched.
+    That's a good way of eliminating most persons early on in the query pipeline, which can greatly reduce the overall
+    memory usage of a query (as aggregation by version happens in-memory).
+    """
     DIRECT_ON_EVENTS = auto()
+    """Get person property from the events table (persons-on-events v1), selecting the latest version of the person."""
     DIRECT_ON_EVENTS_WITH_POE_V2 = auto()
-    DIRECT_ON_PERSONS = auto()
+    """Get person property from the events table (persons-on-events v2), selecting the latest version of the person."""
```

Comment:
I've added this documentation, please let me know if this is not fully aligned with reality @neilkakkar!

Reply:
great work!